cmd/admin-revoker/main.go

```diff
@@ -211,7 +240,129 @@ func (r *revoker) revokeByReg(ctx context.Context, regID int64, reasonCode revoc
 }
 
 func (r *revoker) revokeMalformedBySerial(ctx context.Context, serial string, reasonCode revocation.Reason) error {
-	return r.revokeCertificate(ctx, core.Certificate{Serial: serial}, reasonCode)
+	return r.revokeCertificate(ctx, core.Certificate{Serial: serial}, reasonCode, false)
+}
+
+// blockByPrivateKey blocks future issuance for certificates with a public key
+// matching the SPKI hash of the provided private key. Reason code must be 1
+// (Key Compromise). Keys passed to this function will be validated as a pair
+// before any actions are taken. This method does not revoke any certificates
```

Comment:
```suggestion
// (Key Compromise). Keys passed to this function will be validated as a pair
// before any actions are taken. This method does not revoke any certificates
```

"validated as a pair" is a little confusing since the function signature only shows a private key. How about: "The SubjectPublicKeyInfo will be generated from the PublicKey embedded in `privateKey`, after verifying that the PublicKey is actually a match for the private key. See https://blog.hboeck.de/archives/888-How-I-tricked-Symantec-with-a-Fake-Private-Key.html for an example of private keys embedding a mismatched public key."

Reply:
Awesome, thanks!